docs/50_Writable_views.md

```diff
@@ -0,0 +1,194 @@
+
+Here is the view definition:
+
+
+```js title=view
+// View
+module.exports = {
+  name: 'orders-details',
+  source: 'orders',
+  type: 'view',
+  enableLookup: true, // Enable the lookup feature
+  pipeline: [
+    {
+      $lookup: {
+        from: 'riders',
+        localField: 'id_rider',
+        foreignField: '_id',
+        as: 'rider',
+        pipeline: [
+          {
+            $project: {
+              _id: 0, // To hide the _id field in the view
+              value: {
+                $toObjectId: '$_id', // Foreign key
+              },
+              label: {
+                $toString: {
+                  $concat: ['$name', ' ', '$surname'], // the aggregated value
+                },
+              },
+            },
+          },
+        ],
+      },
+    },
+    {
+      $unwind: {
+        path: '$rider',
+        preserveNullAndEmptyArrays: true,
+      },
+    },
+    {
+      $project: {
+        _id: 1,
+        rider: 1,
+        items: 1,
+        __STATE__: 1,
+      },
+    },
+  ],
+}
+```
+
+And here the colleciton definition:
+
+```js title=collection
+// Collection
+module.exports = {
+  name: 'orders-details',
+  endpointBasePath: '/orders-details-endpoint',
+  defaultState: 'PUBLIC',
+  fields: [
+    {
+      name: '_id',
+      type: 'ObjectId',
+      required: true,
+    },
+    {
+      name: '__STATE__',
+      type: 'string',
+      description: 'The state of the document',
+      required: true,
+    },
+    {
+      name: 'rider',
+      type: 'RawObject',
+      schema: {
+        properties: {
+          value: {
+            type: 'string',
+            __mia_configuration: {
+              type: 'ObjectId',
+            },
+          },
+          label: { type: 'string' },
+        },
+      },
+      additionalProperties: false,
+      required: true,
+      nullable: false,
+    },
+    {
+      name: 'items',
+      type: 'Array',
+      items: {
+        type: 'string',
+      },
+      description: 'The item to deliver to the customer',
+      required: true,
+      nullable: false,
+    },
+  ],
+  indexes: [
+    {
+      name: '_id',
+      type: 'normal',
+      unique: true,
+      fields: [
+        {
+          name: '_id',
+          order: 1,
+        },
+      ],
+    },
+  ],
+}
+```
+
+This will lead to a this results:
+
+```json
+[
+  {
+    "_id": "6489961b951afe064fd2d0d4",
+    "__STATE__": "PUBLIC",
+    "items": [
+      "Pizza"
+    ],
+    "rider": {
+      "value": "64899515951afe064fd2d0d1",
+      "label": "Foo Bar"
+    }
+  }
+]
+```
+
+As mentioned earlier, the CRUD service will also expose lookup routes. In this case, the lookup field in the view is rider, so the corresponding route will be `/lookup/rider`. Accessing this route will return the following information:
```

Comment:
Only one thing 🤔  Should we add the collection base path to the endpoints where `lookup` are exposed?

E.g.

    /orders-details-endpoint/lookup/rider

Reply:
Absolutely, I forgot this detail, thanks!